WordPress/gutenberg
# Cover Block: Fix regressions

## Description

This PR fixes the following Cover block regressions:
- **Restore background dim styles for old markup**
   Fixes situations where Cover blocks published pre [#35065]() lost their background dim.
- **Sets 100% default background dim for solid colors/gradients and 50% for images**:
   Fixes situations where an Image transformed into Cover gets a 100% opaque overlay by default.

Updated tests:
 - Added E2E test for preserving the background dim when transformed from the Image block
 - ~Updated failing Cover block fixtures~ (see https://github.com/WordPress/gutenberg/pull/36406#issuecomment-969331791)
 
## How has this been tested?

The following scenarios for the Cover block should be verified:
1. When setting background color via the block placeholder, its default opacity is set to 100%
2. When setting image/video via the block placeholder, the default overlay color opacity is set to 50%
3. When transforming from the Image/video block, the default overlay opacity should be 50%
4. Default overlay opacity for blocks published with v11.7.1 and older should persist when viewed via this branch. This can be tested with the following steps:
   1. `git checkout v11.7.1`,
   2. Add the Cover block with Image, do not change the overlay color - it should be black with 50% opacity by default,
   3. Publish the page/post,
   4. `git checkout trunk`,
   5. Hard refresh the published page,
   6. Notice that the overlay color is gone,
   7. `git checkout fix/cover-block-regression`,
   8. Hard refresh again,
   9. Notice the overlay is correct now (as described in step 2).

## Types of changes
Bug fix

## Checklist:
